packages/core/src/authorization.ts

```diff
@@ -0,0 +1,28 @@
+import {Ctx} from "./middleware"
+import {AuthenticatedSessionContext} from "./supertokens"
+import {ZodSchema, infer as zInfer} from "zod"
+
+export type ProtectArgs<T> = {schema?: T; authorize?: boolean | unknown}
```

Comment:
is typing authorize as boolean | unknown correct when the implementation allows it to be an array or string as well? fwiw, I ended up splitting it into two args (authorize: boolean, roles: string[] | string)